runusb

```diff
@@ -122,45 +79,93 @@ class FSTabReader(object):
         self.handle.seek(0)
 
         for line in self.handle:
-            stripped_line = line.strip()
-
-            try:
-                # Split the line into the standard 6 parts; the ValueError
-                # will protect us from ill-formatted lines
-                (device, mountpoint, filesystem, options, _, _) = \
-                    line.split(' ')
-            except ValueError:
-                pass
-
-            if options:
-                options_tuple = tuple(options.split(','))
-            else:
-                options = ()
+            (_, mountpoint, filesystem, _, _, _) = line.split(' ')
 
             yield Mountpoint(
                 mountpoint=mountpoint,
-                device=device,
                 filesystem=filesystem,
-                options=options,
             )
 
     def watch(self, timeout=None):
         _, _, changed = select.select([], [], [self.handle], timeout)
 
         if changed:
-            LOGGER.debug("Detected change in %s", self.path)
+            LOGGER.debug("Detected change in procfile")
 
         return bool(changed)
 
 
+def update_robot(path):
+    command = [
+        'sb-update',
+        path
+    ]
+    return subprocess.Popen(command, stdin=subprocess.DEVNULL)
+
+
+def run_robot_code(path):
+    command = [
+        'systemd-nspawn',
+        '--read-only',
+        '--private-network'
+    ]
+
+    # Specify the root directory for the nspawn container
+    command.extend(('--directory', SPAWN_IMAGE))
+
+    # Bring in any requested bind mounts
+    command.extend(('--bind', ROBOTD_LOCATION))
+
+    # Mount a tmpfs as /tmp
+    command.extend(('--tmpfs', '/tmp'))
+
+    # Bind-mount the actual path
+    command.extend(('--bind', path))
+
+    # Attach a machine ID, use the path as the ID
+    command.extend(('--machine', get_machinename(path)))
+
+    # Run the autorun file with bash -c
+    command.extend(('/bin/bash', '-c'))
+
+    # cd to the mountpoint and run the autorun file
+    command.append(
+        'cd "{mountpoint}" ; script -ec "python3 {autorun}" -f log.txt'.format(
+            mountpoint=path,
+            autorun=ROBOT_FILE,
+        ),
+    )
+
+    return subprocess.Popen(command, stdin=subprocess.DEVNULL)
+
+
+def kill_robot_process(path, process):
+    # With the mountpoint now missing the only sensible thing to do is
+    # to die horribly.
+    print(get_machinename(path))
```

Comment:
Do we still need this print?

Reply:
Unlikely. Although I think factoring out those / using `logging` should probably be done as a different PR

Reply:
Agreed